src/_pytest/_code/code.py

```diff
@@ -697,26 +697,79 @@ def getrepr(
         )
         return fmt.repr_excinfo(self)
 
+    def _stringify_exception(self, exc: BaseException) -> str:
+        return "\n".join(
+            [
+                str(exc),
+                *getattr(exc, "__notes__", []),
+            ]
+        )
+
     def match(self, regexp: Union[str, Pattern[str]]) -> "Literal[True]":
         """Check whether the regular expression `regexp` matches the string
         representation of the exception using :func:`python:re.search`.
 
         If it matches `True` is returned, otherwise an `AssertionError` is raised.
         """
         __tracebackhide__ = True
-        value = "\n".join(
-            [
-                str(self.value),
-                *getattr(self.value, "__notes__", []),
-            ]
-        )
+        value = self._stringify_exception(self.value)
         msg = f"Regex pattern did not match.\n Regex: {regexp!r}\n Input: {value!r}"
         if regexp == value:
             msg += "\n Did you mean to `re.escape()` the regex?"
         assert re.search(regexp, value), msg
         # Return True to allow for "assert excinfo.match()".
         return True
 
+    def _group_contains(
+        self,
+        exc_group: BaseExceptionGroup[BaseException],
+        expected_exception: Union[Type[BaseException], Tuple[Type[BaseException], ...]],
+        match: Union[str, Pattern[str], None],
+        recursive: bool = False,
+    ) -> bool:
+        """Return `True` if a `BaseExceptionGroup` contains a matching exception."""
+        for exc in exc_group.exceptions:
+            if recursive and isinstance(exc, BaseExceptionGroup):
+                if self._group_contains(exc, expected_exception, match, recursive):
+                    return True
+            if not isinstance(exc, expected_exception):
+                continue
+            if match is not None:
+                value = self._stringify_exception(exc)
+                if not re.search(match, value):
+                    continue
+            return True
+        return False
+
+    def group_contains(
+        self,
+        expected_exception: Union[Type[BaseException], Tuple[Type[BaseException], ...]],
+        match: Union[str, Pattern[str], None] = None,
+        recursive: bool = False,
```

Comment:
Rather than `recursive: bool = False`, I'd propose `depth: Optional[int] = None`.

This will make it easy to check that the exception is nested at a certain depth, but default to allowing any depth (`None`) so that loose checks can be as easy as possible in the common case that users don't really care about _exactly_ how deeply their exception groups are nested.  I'd treat the caught exception as depth=0, and therefore validate that the user passed depth >= 1 if it's non-None.